json.md

```diff
@@ -0,0 +1,415 @@
+
+dev_dependencies:
+  # Your other dev_dependencies here
+  build_runner: ^0.6.1
+  json_serializable: ^0.3.0
+```
+
+After declaring the needed dependencies, synchronize your packages. In IntelliJ
+IDEA, click the _Packages get_ link on the top of an opened `pubspec.yaml` file.
+Alternatively, you can do synchronize packages by running `flutter packages get`
+from the command line in your projects' root folder.
+
+Next,  define some build actions.
+
+#### Defining the build actions
+
+_Build actions_ are a way of running _builders_ in our project's source files.
+
+Our build action has the following parameters:
+
+* **builder**, which in our case is a `jsonPartBuilder`
+* **package name** which is the specific package this build action will be
+  performed on
+* **inputs**, which are files the build action is performed on.
+
+We'll create our build action file in `<project_root>/tool/build_actions.dart`.
+
+**tool/build_actions.dart**
+
+<!-- skip -->
+{% prettify dart %} 
+import 'package:build_runner/build_runner.dart';
+import 'package:json_serializable/json_serializable.dart';
+
+List<BuildAction> get buildActions => [
+      new BuildAction(
+        // The builder to run on our source files.
+        jsonPartBuilder(),
+
+        // The name of the current package, usually the same as the name 
+        // for your Flutter project. For example, if you created a new
+        // project with "flutter run hello", your package name here would
+        // be 'hello'.
+        '[[highlight]]<your_package_name_here>[[/highlight]]',
+
+        // All of the files this `BuildAction` should target when run.
+        inputs: const [
+          'lib/*.dart',
+        ],
+      ),
+    ];
+{% endprettify %}
+
+We are declaring a top-level variable called `buildActions`, which returns a
+`List` containing all build actions in our project. In this case, we only have
+one build action, which generates JSON serialization logic for files in the
+_lib_ folder.
+
+It is good practice to define build actions in the same file, as they are used
+by the _build_ and the _watcher_ files.
+
+Which brings us to the next topic, the build file.
+
+#### Creating the build file
+
+Our build file runs the actions responsible for generating JSON serialization
+logic for our classes. When run, it triggers the build actions, searches for all
+the relevant files, generates the necessary code and then complete.
+
+We'll create the build file in `<project_root>/tool/build.dart`.
+
+**tool/build.dart**
+
+<!-- skip -->
+```dart
+import 'dart:io';
+import 'package:build_runner/build_runner.dart';
+import 'build_actions.dart';
+
+main() async {
+  var result = await build(buildActions, deleteFilesByDefault: true);
+  if (result.status == BuildStatus.failure) {
+    exitCode = 1;
+  }
+}
+```
+
+Now we can generate all the necessary JSON serialization source files by running
+`dart tool/build.dart` in our project's root folder. While the build file is all
```

Comment:
this assumes you have the Dart SDK installed separately from Flutter. Instead, please show how to run the command-line Dart VM that is shipped with the flutter SDK.

Reply:
Here's how to do it: https://groups.google.com/forum/#!topic/flutter-dev/xoS0WSUax8Q